src/lib/components/composites/settings/user-settings/ChangeNameSettings.svelte

```diff
@@ -0,0 +1,114 @@
+<script lang="ts">
+    import { invalidate } from "$app/navigation";
+    import Button from "$lib/components/primitives/button/button.svelte";
+    import { backendService } from "$lib/grpc-api";
+    import { User } from "$lib/model/api/user";
+    import type { ApiError } from "$lib/model/general";
+    import { Schema } from "$lib/schemas";
+    import { toast } from "svelte-sonner";
+    import Input from "../../input/Input.svelte";
+    import ErrorAlert from "../../utils/ErrorAlert.svelte";
+    import SettingsSection from "../SettingsSection.svelte";
+    import { generateFieldMask } from "protobuf-fieldmask";
+
+    interface Props {
+        user: User;
+    }
+
+    const { user }: Props = $props();
+
+    let firstNameInput: Input;
+    let lastNameInput: Input;
+
+    let updateUserError: ApiError | undefined = $state(undefined);
+
+    async function handleSubmit(event: Event) {
+        event.preventDefault();
+
+        const isFirstNameValid = firstNameInput.validate();
+        const isLastNameValid = lastNameInput.validate();
+        if (!(isFirstNameValid && isLastNameValid)) return;
+
+        const userData: Partial<User> = {
+            id: user.id,
+            firstName: firstNameInput.getValue(),
+            lastName: lastNameInput.getValue(),
+        };
+
+        const updatedUser = User.create(userData);
+        const maskPaths = generateFieldMask(userData).filter((path) => path !== "id");
+
+        backendService
+            .updateUser({
+                user: updatedUser,
+                mask: {
+                    paths: maskPaths,
+                },
+            })
+            .response.then(() => {
+                invalidate("data:getCurrentUser");
+                toast("Successfully updated your name.");
+
+                (document.activeElement as HTMLElement)?.blur();
+            })
+            .catch((error) => {
+                updateUserError = { errorTitle: "Something went wrong while updating user." };
+                console.error(error);
+            });
+    }
+</script>
+
+<!--
+@component
+Component for changing the user name setting.
```

Comment:
What are your thoughts on this?
```suggestion
Component for changing the user's name.
```

Reply:
Yes, this is fine. It's shorter and more descriptive.